x-pack/platform/plugins/shared/observability_ai_assistant/server/functions/get_dataset_info/index.ts

```diff
@@ -32,68 +35,87 @@ export function registerGetDatasetInfoFunction({
           index: {
             type: 'string',
             description:
-              'index pattern the user is interested in or empty string to get information about all available indices',
+              'Index pattern the user is interested in. You are allowed to specify multi comma-separated patterns like "index1,index2". If you provide an empty string, all indices will be returned.',
           },
         },
         required: ['index'],
       } as const,
     },
-    async ({ arguments: { index }, messages, chat }, signal) => {
-      const coreContext = await resources.context.core;
+    async ({ arguments: { index: indexPattern }, messages, chat }, signal) => {
+      const content = await getDatasetInfo({ resources, indexPattern, signal, messages, chat });
+      return { content };
+    }
+  );
+}
 
-      const esClient = coreContext.elasticsearch.client;
-      const savedObjectsClient = coreContext.savedObjects.client;
+export async function getDatasetInfo({
+  resources,
+  indexPattern,
+  signal,
+  messages,
+  chat,
+}: {
+  resources: RespondFunctionResources;
+  indexPattern: string;
+  signal: AbortSignal;
+  messages: Message[];
+  chat: FunctionCallChatFunction;
+}) {
+  const coreContext = await resources.context.core;
+  const esClient = coreContext.elasticsearch.client;
+  const savedObjectsClient = coreContext.savedObjects.client;
 
-      let indices: string[] = [];
+  const indices = await getIndicesFromIndexPattern(indexPattern, esClient, resources.logger);
+  if (indices.length === 0 || indexPattern === '') {
+    return { indices, fields: [] };
+  }
 
-      try {
-        const body = await esClient.asCurrentUser.indices.resolveIndex({
-          name: index === '' ? ['*', '*:*'] : index.split(','),
-          expand_wildcards: 'open',
-        });
-        indices = [
-          ...body.indices.map((i) => i.name),
-          ...body.data_streams.map((d) => d.name),
-          ...body.aliases.map((d) => d.name),
-        ];
-      } catch (e) {
-        indices = [];
-      }
+  try {
+    const { fields, stats } = await getRelevantFieldNames({
+      index: indices,
+      messages,
+      esClient: esClient.asCurrentUser,
+      dataViews: await resources.plugins.dataViews.start(),
+      savedObjectsClient,
+      signal,
+      chat,
+    });
+    return { indices, fields, stats };
+  } catch (e) {
+    resources.logger.error(`Error getting relevant field names: ${e.message}`);
+    return { indices, fields: [] };
+  }
+}
 
-      if (index === '') {
-        return {
-          content: {
-            indices,
-            fields: [],
-          },
-        };
-      }
+async function getIndicesFromIndexPattern(
+  indexPattern: string,
+  esClient: IScopedClusterClient,
+  logger: Logger
+) {
+  try {
+    const name = indexPattern === '' ? ['*', '*:*'] : indexPattern.split(',').map((i) => `*${i}*`);
```

Comment:
do you want to add remote clusters here as well automatically?

Reply:
Fixed